pkg/login/social/socialimpl/service.go

```diff
@@ -71,12 +71,18 @@ func ProvideService(cfg *setting.Cfg,
 			sec := cfg.Raw.Section("auth." + name)
 
 			settingsKVs := convertIniSectionToMap(sec)
+
 			info, err := connectors.CreateOAuthInfoFromKeyValues(settingsKVs)
 			if err != nil {
 				ss.log.Error("Failed to create OAuthInfo for provider", "error", err, "provider", name)
 				continue
 			}
 
+			// Workaround for moving the SkipOrgRoleSync setting to the OAuthInfo struct
+			withOverrides := cfg.SectionWithEnvOverrides("auth." + name)
+			info.Enabled = withOverrides.Key("enabled").MustBool(false)
+			info.SkipOrgRoleSync = sec.Key("skip_org_role_sync").MustBool(false)
```

Comment:
Should we parse these settings as we do above for the other settings using the `connectors.CreateOAuthInfoFromKeyValues` function?

Reply:
> Should we parse these settings as we do above for the other settings

I don't want to change the current behaviour which is loading everything from the `ini` file + apply the env overrides for the `enabled` + `skip_org_role_sync` only.

The SSO Settings API by default does use the SectionWithEnvOverrides, but it's behind a feature toggle and that's gonna change the behaviour.